### PR TITLE
JS-7 Mosaic Restriction

### PR DESCRIPTION
**Pull request by fullcircle23**
Completed issue [Elephant] Mosaic Restriction #76. Please review. Thanks.

### CODE REVIEW

File: sdk-core/src/main/java/io/nem/sdk/model/mosaic/MosaicProperties.java [hunk 1]
```diff
@@ -58,14 +64,45 @@ private MosaicProperties(
...
+     *
+     * @param supplyMutable True supply can change.
+     * @param transferable True mosaic can be transfer.
+     * @param divisibility Decimal place of the mosaic.
+     * @param restrictable True mosaic supports restrictions.
+     * @return Mosaic properties.
+     */
+    public static MosaicProperties create(
+        boolean supplyMutable, boolean transferable, int divisibility, boolean restrictable) {
+        return new MosaicProperties(supplyMutable, transferable, divisibility, restrictable, BigInteger.ZERO);
```
Comment (rg911) [inline sdk-core/src/main/java/io/nem/sdk/model/mosaic/MosaicProperties.java]: We should make the restrictable by default to `False` I think. As the doc mentioned as well

Comment (fullcircle23) [reply sdk-core/src/main/java/io/nem/sdk/model/mosaic/MosaicProperties.java]: The restrictable is default when not provided. Please refer to this constructor:     
public static MosaicProperties create(
        boolean supplyMutable, boolean transferable, int divisibility) {
        return new MosaicProperties(supplyMutable, transferable, divisibility, false, BigInteger.ZERO);
    }

**Comment by fullcircle23** [comment]
> Overall looks good.
> One concern is if you look at the new mosaic definition schema on catbuffer / catapult-rest payload, there is no more `mosaicProperties` anymore. Instead it's replaced by `flags`, `divisibility` and `duration`. The refactoring on TS hasn't been done yet, but I think this could fit in this PR perhaps?

Yes. Will fit the change in this PR.

**Comment by fullcircle23** [comment]
> > Overall looks good.
> > One concern is if you look at the new mosaic definition schema on catbuffer / catapult-rest payload, there is no more `mosaicProperties` anymore. Instead it's replaced by `flags`, `divisibility` and `duration`. The refactoring on TS hasn't been done yet, but I think this could fit in this PR perhaps?
> 
> Yes. Will fit the change in this PR.

Completed Java sdk changes for catbuffer mosaic definition schema changes in https://github.com/nemtech/catbuffer/commit/9da899c3d9569cf6faf3921c9780f2355b636fa6

File: sdk-core/src/main/java/io/nem/sdk/model/blockchain/BlockDuration.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    /**
+     * Returns the number of blocks from height it will be active
+     *
+     * @return the number of blocks from height it will be active
+     */
+    public long getDuration() {
+        return duration;
+    }
```
Comment (rg911) [inline sdk-core/src/main/java/io/nem/sdk/model/blockchain/BlockDuration.java]: Do we really need this model? Duration holds BigInt value only. In TS SDK it is used as a variable only with the type of UInt64

**Pull request merged**